lib/jira/resource/user.rb

```diff
@@ -13,7 +14,7 @@ def self.singular_path(client, key, prefix = '/')
       # This method is a bit of a hack. There is no way to get a list of all users in Jira, so we concatenate all of the project IDs
       # and use the assignable users endpoint to get a list of users that can be assigned to all projects.
       def self.all(client)
-        project_list = client.Project.all.map(&:key).join(',')
+        project_list = client.Project.all.map(&:key).take(SMALLER_SAMPLE_SIZE).join(',')
         response = client.get("#{client.options[:rest_base_path]}/user/assignable/multiProjectSearch?projectKeys=#{project_list}&maxResults=1000")
```

Comment:
Should we change `maxResults` in the get request from 1000 to `SMALLER_SAMPLE_SIZE` as well? I vaguely remember working on some Jira bug where it turned out that Jira changed the max results it would return from 1000 to 100 without documenting that and it was causing weird problems, so it seems like their API just has a problem dealing with 100+ results for anything.

Reply:
@arincon9 what do you think about the smaller sample size for `maxResults` here? Have we tested that we can get more than 100 users? Or if so, what about >100 users per project? It would be good to understand the limits here, because it feels like we're kind of kicking the can down the road and may run into another limit.

That said, what we have here gets us over the current hurdle, and other than the above questions it looks good to me.